### ISSUE TITLE
[swarm] routing mesh not working as expected on secundary manager node

### ISSUE THREAD

**Issue by rogaha** [docker/docker#25434]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 23:54:00 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 23:54:00 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 9
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge overlay null host
Swarm: active
 NodeID: 38o99pjjg4zdixi4lkkyhbeuj
 Is Manager: true
 ClusterID: 3dda9428jynvr4014diorlz0l
 Managers: 2
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot interval: 10000
  Heartbeat tick: 1
  Election tick: 3
 Dispatcher:
  Heartbeat period: 5 seconds
 CA configuration:
  Expiry duration: 3 months
 Node Address: 192.168.99.101
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.16-boot2docker
Operating System: Boot2Docker 1.12.0 (TCL 7.2); HEAD : e030bab - Fri Jul 29 00:29:14 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 995.9 MiB
Name: master1
ID: WHIS:VMFW:ABXS:YJIO:AQHB:ZTI4:UDT4:TVRC:WE4V:C2PW:E6TA:7U3P
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 46
 Goroutines: 164
 System Time: 2016-08-05T10:16:51.778932882Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
5 VirtualBox VMs cluster

**Steps to reproduce the issue:**
1. docker network create -d overlay whoami-net
2. docker service create --name service1 --network whoami-net -p 8000 jwilder/whoami
3. docker service scale service1=3

master2 node logs -- 
[logs2.txt](https://github.com/docker/docker/files/403582/logs2.txt)

**Describe the results you received:**
```
docker@master1:~$ docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
1vxqha68ascue14y9wguzb1yi    node3     Ready   Active
38o99pjjg4zdixi4lkkyhbeuj *  master1   Ready   Active        Leader
5wl3mpokt7vqjw9jwmf7155o4    master2   Ready   Active        Reachable
a9smf5upud673q1sot2gjrco1    node1     Ready   Active
agxwdu4fm7wye2yqqry3cgnvj    node2     Ready   Active
docker@master1:~$
docker@master1:~$ docker node inspect 5wl3mpokt7vqjw9jwmf7155o4
[
    {
        "ID": "5wl3mpokt7vqjw9jwmf7155o4",
        "Version": {
            "Index": 16
        },
        "CreatedAt": "2016-08-05T09:57:25.91211974Z",
        "UpdatedAt": "2016-08-05T09:57:26.295315567Z",
        "Spec": {
            "Role": "manager",
            "Availability": "active"
        },
        "Description": {
            "Hostname": "master2",
            "Platform": {
                "Architecture": "x86_64",
                "OS": "linux"
            },
            "Resources": {
                "NanoCPUs": 1000000000,
                "MemoryBytes": 1044246528
            },
            "Engine": {
                "EngineVersion": "1.12.0",
                "Labels": {
                    "provider": "virtualbox"
                },
                "Plugins": [
                    {
                        "Type": "Network",
                        "Name": "bridge"
                    },
                    {
                        "Type": "Network",
                        "Name": "host"
                    },
                    {
                        "Type": "Network",
                        "Name": "null"
                    },
                    {
                        "Type": "Network",
                        "Name": "overlay"
                    },
                    {
                        "Type": "Volume",
                        "Name": "local"
                    }
                ]
            }
        },
        "Status": {
            "State": "ready"
        },
        "ManagerStatus": {
            "Reachability": "reachable",
            "Addr": "192.168.99.102:2377"
        }
    }
]
docker@master1:~$ curl http://192.168.99.102:30000/
curl: (7) Failed to connect to 192.168.99.102 port 30000: Connection refused
```
**Describe the results you expected:**
```
docker@master1:~$ curl http://192.168.99.102:30000/
I'm af88259a7bd8
docker@master1:~$ curl http://192.168.99.102:30000/
I'm 51d04367776d
docker@master1:~$ curl http://192.168.99.102:30000/
I'm ea05c707cc49
docker@master1:~$ curl http://192.168.99.102:30000/
I'm af88259a7bd8
```
**Additional information you deem important (e.g. issue happens only occasionally):**
All the worker nodes (3) and the master1 node are working as expected (load balancing the requests). So, there is just one node failing (manager node).

**Comment by fordodone** [comment]
I experience the same thing with a 3 node swarm (all managers). The service only listens 
(netstat shows port LISTEN) on the node where the container gets scheduled.

**Comment by rogaha** [comment]
This issue seems to be solved on `1.12.2-rc1`. I wasn't able to reproduce it anymore.
